demo/src/pages/for.astro

```diff
@@ -10,95 +11,56 @@ import { For, Iterate, iterate } from '@astropub/flow'
 	<h2>
 		Usage with <code>iterate()</code>
 	</h2>
-	<pre><code
->&lt;ul&gt;&lcub;iterate([
+	<pre><code>&lt;ul&gt;&lcub;iterate([
 	&lcub; title: 'This is the first item.' &rcub;,
 	&lcub; title: 'This is the second item.' &rcub;,
 	&lcub; title: 'This is the last item.' &rcub;,
 ], item =&gt; (
 	&lt;li&gt;&lcub;item.title&rcub; (method)&lt;/li&gt;
 )&rcub;&lt;/ul&gt;</code></pre>
-	<ul>{iterate([
-		{ title: 'This is the first item.' },
-		{ title: 'This is the second item.' },
-		{ title: 'This is the last item.' },
-	], item => (
-		<li>{item.title} (method)</li>
-	))}</ul>
-	<br />
-	<h2>
-		Usage with <code>&lt;Iterate&gt;</code>
-	</h2>
-	<pre><code
->&lt;ul&gt;
-	&lt;Iterate of=&lcub;[
-		&lcub; title: 'This is the first item.' &rcub;,
-		&lcub; title: 'This is the second item.' &rcub;,
-		&lcub; title: 'This is the last item.' &rcub;,
-	]&rcub;&gt;&lcub;item =&gt; (
-		&lt;li&gt;&lcub;item.title&rcub; (component)&lt;/li&gt;
-	)&rcub;&lt;/Iterate&gt;
-&lt;/ul&gt;</code></pre>
 	<ul>
-		<Iterate of={[
-			{ title: 'This is the first item.' },
-			{ title: 'This is the second item.' },
-			{ title: 'This is the last item.' },
-		]}>{item => (
-			<li>{item.title} (component)</li>
-		)}</Iterate>
+		{
+			iterate(
+				[
+					{ title: "This is the first item." },
+					{ title: "This is the second item." },
+					{ title: "This is the last item." },
+				],
+				(item) => <li>{item.title} (method)</li>
+			)
+		}
 	</ul>
 	<br />
 	<h2>
-		Usage with <code>iterate()</code> and a Generator
+		Usage with <code>&lt;For&gt;</code>
 	</h2>
-	<pre><code
->&lt;ul&gt;&lcub;iterate([
-	&lcub; title: 'This is the first item.' &rcub;,
-	&lcub; title: 'This is the second item.' &rcub;,
-	&lcub; title: 'This is the last item.' &rcub;,
-], function * (item) &lcub;
-	yield &lt;li&gt;&lcub;item.title&rcub; (method + generator)&lt;/li&gt;
-&rcub;&lt;/ul&gt;</code></pre>
-	<ul>{iterate([
-		{ title: 'This is the first item.' },
-		{ title: 'This is the second item.' },
-		{ title: 'This is the last item.' },
-	], function * (item) {
-		yield <li>{item.title} (method + generator)</li>
-	})}</ul>
-	<br />
-	<h2>
-		Usage with <code>&lt;For&gt;</code> and a Generator
-	</h2>
-	<pre><code
->&lt;ul&gt;
+	<pre><code>&lt;ul&gt;
 	&lt;For of=&lcub;[
 		&lcub; title: 'This is the first item.' &rcub;,
 		&lcub; title: 'This is the second item.' &rcub;,
 		&lcub; title: 'This is the last item.' &rcub;,
-	]&rcub;&gt;&lcub;function * (item) &lcub;
-		yield &lt;li&gt;&lcub;item.title&rcub; (component)&lt;/li&gt;
```

Comment:
Why are all of the generators removed?

Reply:
You can't really use generators anymore. I mean you can put them in there but yield won't work properly. A generator is being used behind the scenes anyway. I'm not sure you looked at how iterate works but. It uses a generator.

Reply:
When I test the current version, the generators are working.

Reply:
> yield won't work properly

That could be because you tried doing it within an Astro file. Astro has changed how components work a few times. Astro used to render components like a stream, skipping slots unless they were set to be rendered, but recently they started rendering components really poorly from the inside out like React, including rendering slots regardless if they were applied. I was told they would be fixed, which is why the current version supports both.

Reply:
I changed `iterate()` so that it can work with just generators.